subprojects/core/src/main/java/org/gradle/execution/plan/DefaultExecutionPlan.java

```diff
@@ -250,7 +251,7 @@ private void requireWithDependencies(Node node) {
     }
 
     public void determineExecutionPlan() {
-        List<NodeInVisitingSegment> nodeQueue = Lists.newArrayList(Iterables.transform(entryTasks, new Function<TaskNode, NodeInVisitingSegment>() {
+        LinkedList<NodeInVisitingSegment> nodeQueue = Lists.newLinkedList(Iterables.transform(entryTasks, new Function<TaskNode, NodeInVisitingSegment>() {
```

Comment:
Why a LinkedList here instead of an ArrayDeque?

Reply:
Because this Queue is not used as one, we're inserting to the middle, too: https://github.com/gradle/gradle/pull/9558/files/5c815e721f21cc69af73781cef2d8b3ee2ff498d#diff-756e465001dc6712187ece0fabd15007R341